internal/dataplane/kongstate/kongstate.go

```diff
@@ -191,17 +191,30 @@ func (ks *KongState) FillConsumerGroups(_ logr.Logger, s store.Storer) {
 	}
 }
 
+func servicesAsObjects(svc *corev1.Service, _ int) client.Object {
```

Comment:
This function looks weird since it has an ignored parameter. Because it is only used inside the `lo.Map` below, can we define it as an inline function or add comments on it?

Reply:
I think it is because the serviceAsObjects also be used in the following positions.

https://github.com/Kong/kubernetes-ingress-controller/pull/5399/files#diff-8150f615be951138f2aa0d9663e50ce02a3e05a7564e38d01f818c6645e7680fR238

https://github.com/Kong/kubernetes-ingress-controller/pull/5399/files#diff-8150f615be951138f2aa0d9663e50ce02a3e05a7564e38d01f818c6645e7680fR247

Reply:
I think it is reasonable to extract it out since it is used twice, but it will be clearer if we add comments on it.

Reply:
Not quite sure what a good comment for it would be (is there some term for why Go can't accept an array of object satisfying an interface when it wants variable arity args of interface?), but it has one now :shrug: